lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenDefinitionParticipant.java

```diff
@@ -62,27 +75,53 @@ public void findDefinition(IDefinitionRequest request, List<LocationLink> locati
 					relativeFile = new File(relativeFile, Maven.POMv4);
 				}
 				if (relativeFile.isFile()) {
-					locations.add(toLocation(relativeFile, (DOMElement)parentNode));
+					locations.add(toLocationNoRange(relativeFile, (DOMElement)parentNode));
 				}
 				return;
 			} else {
 				File relativeFile = new File(currentFolder.getParentFile(), Maven.POMv4);
 				if (match(relativeFile, dependency)) {
-					locations.add(toLocation(relativeFile, (DOMElement)parentNode));
+					locations.add(toLocationNoRange(relativeFile, (DOMElement)parentNode));
 				}
 				return;
 			}
 		}
 		if (dependency != null) {
 			File artifactLocation = getArtifactLocation(dependency);
-			LocationLink location = toLocation(artifactLocation, element);
+			LocationLink location = toLocationNoRange(artifactLocation, element);
 			if (location != null) {
 				locations.add(location);
 			}
 			return;
 		}
 	}
 
+	private LocationLink findMavenPropertyLocation(String mavenProperty, IDefinitionRequest request) {
+		DOMDocument xmlDocument = request.getXMLDocument();
+		MavenProject project = cache.getLastSuccessfulMavenProject(xmlDocument);
+		MavenProject childProj = project;
+		while (project != null && project.getProperties().containsKey(mavenProperty)) {
+			childProj = project;
+			project = project.getParent();
+		}
+		
+		DOMNode propertyDeclaration = null;
+		// The substring is an ugly hack to get the path of the xmlDocument
+		if (childProj.getFile().toURI().getPath().equals(xmlDocument.getDocumentURI().substring(5))) {
```

Comment:
@mickaelistria any suggestion on how to better compare that the xml document & maven project's file are equal? I had to remove the `"file:"` part of the documentURI string.

Reply:
would `childProj.getFile().toURI(),toString().equals(xmlDocument.getDocumentURI))` work?

Reply:
Works perfectly :)!